earthquakes.py

```diff
@@ -11,47 +12,58 @@ def get_data():
         params={
-            'starttime': "2000-01-01",
-            "maxlatitude": "58.723",
-            "minlatitude": "50.008",
-            "maxlongitude": "1.67",
-            "minlongitude": "-9.756",
-            "minmagnitude": "1",
-            "endtime": "2018-10-11",
-            "orderby": "time-asc"}
+            'starttime': "2000-01-01", #beginning date of query
+            #Use a rectangle to cover the map (min-max lat and min-max long)
+            "maxlatitude": "58.723", #UK maximum latitude to query
+            "minlatitude": "50.008", #UK minimum latitude to query
+            "maxlongitude": "1.67", #Upper value of UK longitude to query
+            "minlongitude": "-9.756", #Lower value of UK longtitude to query
+            "minmagnitude": "1", #Looking for earthquakes with a higher magnitude than 1
+            "endtime": "2018-10-11", #End of date range we are looking into
+            "orderby": "time-asc"} #want time ascending dates
     )
 
     # The response we get back is an object with several fields.
     # The actual contents we care about are in its text field:
     text = response.text
-    # To understand the structure of this text, you may want to save it
-    # to a file and open it in VS Code or a browser.
-    # See the README file for more information.
-    ...
-
-    # We need to interpret the text to get values that we can work with.
-    # What format is the text in? How can we load the values?
-    return ...
+    
+    #write to a json file
+    with open('earthquakes_data.json', 'w') as earthquakes_data_out:
+        earthquakes_data_out.write(json.dumps(text,indent=4))
+        
+    #parse text variable using json loads
+    data = json.loads(text)
+    return data
 
 def count_earthquakes(data):
     """Get the total number of earthquakes in the response."""
-    return ...
+    #upon inspecting the JSON file, it is clear that the metadata ontains the count (120)
+    return data["metadata"]["count"] #access metadata dictionary within data dictionary to find count key
 
 
 def get_magnitude(earthquake):
     """Retrive the magnitude of an earthquake item."""
-    return ...
+    #We want to access the magnitude for a given earthquake
+    return earthquake["properties"]["mag"]
 
 
 def get_location(earthquake):
     """Retrieve the latitude and longitude of an earthquake item."""
     # There are three coordinates, but we don't care about the third (altitude)
-    return ...
+    return earthquake["geometry"]["coordinates"][0:1]
 
 
 def get_maximum(data):
     """Get the magnitude and location of the strongest earthquake in the data."""
-    ...
+    #extract magnitudes
+    mags = [get_magnitude(earthquakes) for earthquakes in data]
+    #find maximum magnitude
+    max_magnitude = max(mags)
+    mag_index = mags.index( max_magnitude)
+    #get locations list
+    locs = [get_location(earthquakes) for earthquakes in data]
+    max_location = locs[mag_index]
```

Comment:
Since you only need the location of one earthquake and you already know its index, you don't need to construct the list of all locations. You could instead
```suggestion
    max_location = get_location(data[mag_index])
```
which would be a bit more efficient.